bindings.go

```diff
@@ -3660,16 +3715,97 @@ func TableDescriptionGetColumnName(desc TableDescription, index IdxT) string {
 	return C.GoString(cName)
 }
 
+// --------------------------------------------------- //
+// Arrow Interface (the new C API bindings)
+// --------------------------------------------------- //
+
+// NewArrowSchema creates a new ArrowSchema from an array of DuckDB logical types and column names.
+func NewArrowSchema(options *ArrowOptions, types []LogicalType, names []string) (*arrow.Schema, error) {
+	cTypes := allocLogicalTypes(types)
+	defer Free(unsafe.Pointer(cTypes))
+
+	cNames := allocNames(names)
+	defer Free(unsafe.Pointer(cNames))
+
+	arr := C.calloc(1, C.sizeof_struct_ArrowSchema)
+	defer func() {
+		cdata.ReleaseCArrowSchema((*cdata.CArrowSchema)(arr))
+		C.free(arr)
+	}()
+
+	ed := C.duckdb_to_arrow_schema(options.data(), cTypes, cNames, C.idx_t(len(names)), (*C.struct_ArrowSchema)(arr))
+	errData := ErrorData{Ptr: unsafe.Pointer(ed)}
+	defer DestroyErrorData(&errData)
+	if ErrorDataHasError(errData) {
+		return nil, errors.New(ErrorDataMessage(errData))
+	}
+	return cdata.ImportCArrowSchema((*cdata.CArrowSchema)(arr))
+}
+
+// DataChunkToArrowArray converts a DuckDB DataChunk to an Arrow RecordBatch using the provided ArrowOptions and schema.
+func DataChunkToArrowArray(options *ArrowOptions, schema *arrow.Schema, chunk *DataChunk) (arrow.RecordBatch, error) {
+	arr := C.calloc(1, C.sizeof_struct_ArrowArray)
+	defer func() {
+		cdata.ReleaseCArrowArray((*cdata.CArrowArray)(arr))
+		C.free(arr)
+	}()
+
+	ed := C.duckdb_data_chunk_to_arrow(options.data(), chunk.data(), (*C.struct_ArrowArray)(arr))
+	errData := ErrorData{Ptr: unsafe.Pointer(ed)}
+	defer DestroyErrorData(&errData)
+	if ErrorDataHasError(errData) {
+		return nil, errors.New(ErrorDataMessage(errData))
+	}
+	return cdata.ImportCRecordBatchWithSchema((*cdata.CArrowArray)(arr), schema)
+}
+
+// SchemaFromArrow converts an Arrow Schema to a DuckDB ArrowConvertedSchema using the provided Connection.
+func SchemaFromArrow(conn Connection, schema *arrow.Schema) (*ArrowConvertedSchema, error) {
```

Comment:
Let's return `(ArrowConvertedSchema, ErrorData)` here to avoid a heap allocation.

Reply:
done